### PR TITLE
map backend error

### PR DESCRIPTION
**Pull request by afik**
Changelog:
Map backend error so it will have same error as xendit.js

### CODE REVIEW

File: xendit-android/src/main/java/com/xendit/TokenBroadcastReceiver.java [hunk 1]
```diff
@@ -32,7 +32,11 @@ public void onReceive(Context context, Intent intent) {
             if (authentication.getStatus().equals("VERIFIED")) {
                 tokenCallback.onSuccess(new Token(authentication));
             } else {
-                tokenCallback.onError(new XenditError(context.getString(R.string.create_token_error_validation), authentication));
+                if (message.contains("AUTHENTICATION_FAILED")) {
```
Comment (0505gonzalez) [inline xendit-android/src/main/java/com/xendit/TokenBroadcastReceiver.java]: Why don't we handle this based on status code? If it's 200, we return the authentication object. if it's anything else and it's a json object, we return the error object with code and original message?

File: xendit-android/src/main/java/com/xendit/Models/XenditError.java [hunk 2]
```diff
@@ -1,35 +1,48 @@
...
-    public XenditError(String error, Authentication authentication) {
-        this(error);
-        this.authentication = authentication;
+    public XenditError(NetworkError networkError) {
+        try {
+            this.errorCode =  networkError.errorResponse.getString("error_code");
+            this.errorMessage = networkError.errorResponse.getString("message");
+        } catch (JSONException e) {
+            e.printStackTrace();
+            this.errorCode = null;
```
Comment (0505gonzalez) [inline xendit-android/src/main/java/com/xendit/Models/XenditError.java]: Why don't we make this default to SERVER_ERROR?

**Comment by 0505gonzalez** [comment]
Otherwise LGTM

File: xendit-android/src/main/java/com/xendit/Models/XenditError.java (hunk 2 above)
Comment (afik) [inline xendit-android/src/main/java/com/xendit/Models/XenditError.java]: Oh you right..

**Pull request merged**